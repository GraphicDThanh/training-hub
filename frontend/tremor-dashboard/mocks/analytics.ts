export const SALES_BY_COUNTRY = [
  {
    id: "us",
    flag: "/assets/images/flags/US.webp",
    country: "United State",
    sales: 2500,
    value: 230900,
    bounce: 29.9,
  },
  {
    id: "ge",
    flag: "/assets/images/flags/DE.webp",
    country: "Germany",
    sales: 3900,
    value: 440000,
    bounce: 40.22,
  },
  {
    id: "gb",
    flag: "/assets/images/flags/GB.webp",
    country: "Great Britain",
    sales: 1400,
    value: 190700,
    bounce: 23.44,
  },
  {
    id: "br",
    flag: "/assets/images/flags/BR.webp",
    country: "Brasil",
    sales: 562,
    value: 143960,
    bounce: 32.14,
  },
  {
    id: "au",
    flag: "/assets/images/flags/AU.webp",
    country: "Australia",
    sales: 400,
    value: 143960,
    bounce: 56.83,
  },
];

export const STATISTICAL_DATA = [
  {
    id: "1",
    type: "Bookings",
    amount: 281,
    amountChange: 55,
    duration: "than last week",
    amountChangeType: 0,
  },
  {
    id: "2",
    type: "Today's Users",
    amount: 2300,
    amountChange: 3,
    duration: "than last month",
    amountChangeType: 0,
  },
  {
    id: "3",
    type: "Revenue",
    amount: 34,
    amountChange: 1,
    duration: "than yesterday",
    amountChangeType: 0,
  },
  {
    id: "4",
    type: "Followers",
    amount: 91,
    amountChange: 0,
    duration: "Just updated",
    amountChangeType: 0,
  },
];

export const ANALYTIC_INFO = [
  {
    id: "1",
    photo: "/assets/images/analytics/analytics-info-1.webp",
    name: "Cozy 5 Stars Apartment",
    description: `The place is close to Barceloneta Beach and bus stop just 2 min by walk and near to "Navig" where you can enjoy the main night life in Barcelona.`,
    price: 899,
    location: "Barcelona, Spain",
  },
  {
    id: "2",
    photo: "/assets/images/analytics/analytics-info-2.webp",
    name: "Office Studio",
    description: `The place is close to Metro Station and bus stop just 2 min by walk and near to "Navig" where you can enjoy the night life in London, UK.`,
    price: 1.119,
    location: "London, UK",
  },
  {
    id: "3",
    photo: "/assets/images/analytics/analytics-info-3.webp",
    name: "Beautiful Castle",
    description: `The place is close to Metro Station and bus stop just 2 min by walk and near to "Navig" where you can enjoy the main night life in Milan.`,
    price: 459,
    location: "Milan, Italy",
  },
];

export const WEBSITE_CHART = {
  id: "view-campaign",
  display: "Website Views",
  desc: "Last Campaign Performance",
  modified: "campaign sent 2 days ago",
  data: [
    {
      description: "M",
      sales: 50,
    },
    {
      description: "T",
      sales: 20,
    },
    {
      description: "W",
      sales: 10,
    },
    {
      description: "T",
      sales: 22,
    },
    {
      description: "F",
      sales: 50,
    },
    {
      description: "S",
      sales: 10,
    },
    {
      description: "S",
      sales: 40,
    },
  ],
};

export const LINE_CHART_DATA = [
  {
    id: "sale",
    display: "Daily Sales",
    descValue: "(+15%)",
    desc: "increase in today sales.",
    modified: "updated 4 min ago",
    data: [
      {
        date: "Apr",
        "Desktop apps": 50,
      },
      {
        date: "May",
        "Desktop apps": 40,
      },
      {
        date: "Jun",
        "Desktop apps": 300,
      },
      {
        date: "Jul",
        "Desktop apps": 220,
      },
      {
        date: "Aug",
        "Desktop apps": 500,
      },
      {
        date: "Sep",
        "Desktop apps": 250,
      },
      {
        date: "Oct",
        "Desktop apps": 400,
      },
      {
        date: "Nov",
        "Desktop apps": 230,
      },
      {
        date: "Dec",
        "Desktop apps": 500,
      },
    ],
  },
  {
    id: "performance",
    display: "Completed Tasks",
    descValue: "(+15%)",
    desc: "Last Campaign Performance",
    modified: "just updated",
    data: [
      {
        date: "Apr",
        "Mobile apps": 50,
      },
      {
        date: "May",
        "Mobile apps": 40,
      },
      {
        date: "Jun",
        "Mobile apps": 300,
      },
      {
        date: "Jul",
        "Mobile apps": 320,
      },
      {
        date: "Aug",
        "Mobile apps": 500,
      },
      {
        date: "Sep",
        "Mobile apps": 350,
      },
      {
        date: "Oct",
        "Mobile apps": 200,
      },
      {
        date: "Nov",
        "Mobile apps": 230,
      },
      {
        date: "Dec",
        "Mobile apps": 500,
      },
    ],
  },
];

export const ANALITICS_DATA = {
  apartment_statistic: ANALYTIC_INFO,
  daily_sale_statistic: LINE_CHART_DATA[0],
  performance_statistic: LINE_CHART_DATA[1],
  sale_by_country: SALES_BY_COUNTRY,
  sale_statistical: STATISTICAL_DATA,
  web_statistic: WEBSITE_CHART,
};

export const EMPTY_ANALITICS_DATA = {
  apartment_statistic: [],
  daily_sale_statistic: {},
  performance_statistic: {},
  sale_by_country: [],
  sale_statistical: [],
  web_statistic: {},
};
