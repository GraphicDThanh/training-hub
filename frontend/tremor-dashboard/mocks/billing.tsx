// Constants
import { AGGREGATION_TYPE } from "@/constants";
import { BillingInfoData } from "@/types";

export const MOCK_SALARY_DATA = [
  {
    type: AGGREGATION_TYPE.SALARY,
    value: 2000,
  },
  {
    type: AGGREGATION_TYPE.PAYPAL,
    value: 45500,
  },
];

export const MOCK_BILLING_CARD = {
  expire: "11/24",
  cardNumber: "4562 1122 4594 7866",
  holderFullName: "Jack Peterson",
  cardLast4Digit: "7866",
};

export const MOCK_BILLING_INFO: BillingInfoData = {
  id: "1",
  ownerName: "Oliver Liam",
  companyName: "Viking Burrito",
  cardLast4Digit: "1234",
  email: "oliver@burrito.com",
  vat: "FRB1235476",
};

export const MOCK_BILLING: BillingInfoData[] = [
  {
    id: "1",
    ownerName: "Oliver Liam",
    companyName: "Viking Burrito",
    cardLast4Digit: "1234",
    email: "oliver@burrito.com",
    vat: "FRB1235476",
  },
  {
    id: "2",
    ownerName: "Oliver Liam",
    companyName: "Viking Burrito",
    cardLast4Digit: "1234",
    email: "oliver@burrito.com",
    vat: "FRB1235476",
  },
  {
    id: "3",
    ownerName: "Oliver Liam",
    companyName: "Viking Burrito",
    cardLast4Digit: "1234",
    email: "oliver@burrito.com",
    vat: "FRB1235476",
  },
];
