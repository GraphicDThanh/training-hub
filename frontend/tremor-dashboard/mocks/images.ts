export const MOCK_URL_AVATAR =
  "https://demos.creative-tim.com/nextjs-material-dashboard-pro//_next/static/media/product-details-1.f79d1e89.jpg";

export const MOCK_IMAGE_UPLOAD =
  "https://i.ibb.co/2WyPpQQ/0daa97f4b0dbe9a9686984982977da86-1.png";
