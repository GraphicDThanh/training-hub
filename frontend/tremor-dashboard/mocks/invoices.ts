import { Invoice } from "@/types";

export const MOCK_INVOICE_DATA = {
  id: 230220,
  createdAt: "06/03/2019",
  dueAt: "11/03/2019",
  bankInfo: {
    address: "St. Independence Embankment, 050105",
    city: "Bucharest",
    state: "Romania",
    phone: "+4 (074) 1090873",
  },
  customer: {
    fullName: "John Doe",
    address: "4006 Locust View Drive",
    city: "San Francisco",
    stateCode: "CA",
    state: "California",
  },
  products: [
    {
      id: 1,
      productName: "Premium Support",
      quantity: 1,
      price: 9,
    },
    {
      id: 2,
      productName: "Metro Dashboard",
      quantity: 3,
      price: 100,
    },
    {
      id: 3,
      productName: "Parts for service",
      quantity: 2,
      price: 10,
    },
  ],
  totalCost: 345,
};

export const MOCK_INVOICES: Invoice[] = [
  {
    id: 804322,
    createdAt: "March, 21, 2021",
    invoicePrefix: "AR",
    totalCost: 180,
  },
  {
    id: 8908543,
    createdAt: "Apr, 22, 2022",
    invoicePrefix: "AR",
    totalCost: 180,
  },
  {
    id: 415645,
    createdAt: "May, 23, 2023",
    invoicePrefix: "MS",
    totalCost: 180,
  },
  {
    id: 103578,
    createdAt: "Jun, 24, 2024",
    invoicePrefix: "QW",
    totalCost: 180,
  },
  {
    id: 126749,
    createdAt: "Jul, 25, 2025",
    invoicePrefix: "RV",
    totalCost: 180,
  },
];

export const MOCK_INVOICE_HEADER = {
  addressBank: "St. Independence Embankment, 050105",
  cityBank: "Bucharest",
  stateBank: "Romania",
  phoneBank: "+4 (074) 1090873",
  fullName: "John Doe",
  addressCustomer: "4006 Locust View Drive",
  cityCustomer: "San Francisco",
  stateCode: "CA",
  stateCustomer: "California",
};

//addressBank: string;
//cityBank: string;
//stateBank: string;
//phoneBank: string;
//fullName: string;
//addressCustomer: string;
//cityCustomer: string;
//stateCode: string;
//stateCustomer: string;

//id: number;
//createdAt: string;
//orderCode: string; }> & {
// id?: number | undefined;
//createdAt?: string | undefined;
//orderCode?: string | undefined;
