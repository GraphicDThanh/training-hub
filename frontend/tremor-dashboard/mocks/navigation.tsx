// Icons
import { IoPersonSharp, IoReceiptOutline } from "react-icons/io5";

// Types
import { INavigation } from "@/types";

export const NAVIGATION_SETTING: INavigation[] = [
  {
    icon: (
      <IoPersonSharp className="text-primary dark:text-white dark!:hover:text-black" />
    ),
    label: "profile",
    href: "#profile",
  },
  {
    icon: (
      <IoReceiptOutline className="text-primary dark:text-white dark!:hover:text-black" />
    ),
    label: "basic info",
    href: "#basicInfo",
  },
];
