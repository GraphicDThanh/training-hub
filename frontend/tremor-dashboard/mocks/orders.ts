// Types
import { Order, TrackOrderData } from "@/types";

export const MOCK_ORDERS: Order[] = [
  {
    id: 10425,
    createdAt: "2023-11-01T10:20:00+00:00",
    status: 1,
    customer: {
      id: 15,
      avatar:
        "https://demos.creative-tim.com/nextjs-material-dashboard-pro//_next/static/media/team-2.1593fb7f.jpg",
      fullName: "Orlando Imieto",
    },
    revenue: 44.9,
    products: [
      {
        id: 6,
        name: "Phone Case Pink",
        count: 2,
        url: "https://demos.creative-tim.com/nextjs-material-dashboard-pro//_next/static/media/black-chair.b2719b4f.jpeg",
        price: 90,
      },
    ],
  },
  {
    id: 12345,
    createdAt: "2023-11-01T03:20:00+00:00",
    status: 0,
    customer: {
      id: 1,
      avatar:
        "https://demos.creative-tim.com/nextjs-material-dashboard-pro//_next/static/media/team-2.1593fb7f.jpg",
      fullName: "Orlando Imieto",
    },
    revenue: 140.2,
    products: [
      {
        id: 12,
        name: "Office Papers",
        count: 1,
        url: "https://demos.creative-tim.com/nextjs-material-dashboard-pro//_next/static/media/black-chair.b2719b4f.jpeg",
        price: 77,
      },
    ],
  },
  {
    id: 12350,
    createdAt: "2023-11-01T11:03:00+00:00",
    status: 0,
    customer: {
      id: 9,
      avatar:
        "https://demos.creative-tim.com/nextjs-material-dashboard-pro//_next/static/media/team-2.1593fb7f.jpg",
      fullName: "Orlando Imieto",
    },
    revenue: 112.5,
    products: [
      {
        id: 10,
        name: "Water Bottle India",
        count: 3,
        url: "https://demos.creative-tim.com/nextjs-material-dashboard-pro//_next/static/media/black-chair.b2719b4f.jpeg",
        price: 7,
      },
    ],
  },
  {
    id: 12356,
    createdAt: "2023-11-01T09:20:00+00:00",
    status: 2,
    customer: {
      id: 13,
      avatar:
        "https://demos.creative-tim.com/nextjs-material-dashboard-pro//_next/static/media/team-2.1593fb7f.jpg",
      fullName: "Orlando Imieto",
    },
    revenue: 23.9,
    products: [
      {
        id: 10,
        name: "Water Bottle India",
        count: 3,
        url: "https://demos.creative-tim.com/nextjs-material-dashboard-pro//_next/static/media/black-chair.b2719b4f.jpeg",
        price: 34.35,
      },
    ],
  },
  {
    id: 12352,
    createdAt: "2023-11-01T16:05:00+00:00",
    status: 0,
    customer: {
      id: 16,
      avatar:
        "https://demos.creative-tim.com/nextjs-material-dashboard-pro//_next/static/media/team-2.1593fb7f.jpg",
      fullName: "Orlando Imieto",
    },
    revenue: 350,
    products: [
      {
        id: 10,
        name: "Water Bottle India",
        count: 3,
        url: "https://demos.creative-tim.com/nextjs-material-dashboard-pro//_next/static/media/black-chair.b2719b4f.jpeg",
        price: 71,
      },
    ],
  },
];

export const MOCK_INVOICE_ORDER_HEADER = {
  id: 241342,
  createdAt: "2024-01-11T01:48:52Z",
  orderCode: "KF332",
};

export const MOCK_ORDER_SUMMARY = {
  productPrice: 90,
  delivery: 14,
  taxes: 1.95,
};

export const MOCK_TRACK_ORDER: TrackOrderData = {
  id: "#1832412",
  generateOrderAt: "2024-01-11T02:48:52Z",
  deliveredAt: "2024-01-11T02:48:52Z",
  transmittedToCourierAt: "2024-01-11T02:48:52Z",
  generateOrderId: "2024-01-11T02:48:52Z",
};

export const MOCK_ORDER_CONTACT = {
  name: "Leather Wallet",
  url: "https://demos.creative-tim.com/nextjs-material-dashboard-pro//_next/static/media/black-chair.b2719b4f.jpeg",
  date: 18,
};
