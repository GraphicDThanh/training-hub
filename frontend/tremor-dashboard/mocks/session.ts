import { User } from "@/types";

import * as SessionContext from "@/context/session";

export const mockUseSession = (user: User | null = null) => {
  jest.spyOn(SessionContext, "useSession").mockImplementation(() => ({
    user,
    setUser: () => {},
  }));
};
