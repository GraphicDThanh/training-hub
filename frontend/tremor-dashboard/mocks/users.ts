// Types
import { ADMIN_ID } from "@/constants";
import { BirthDay, USER_ROLE, User } from "@/types";

export const MOCK_USERS: User[] = [
  {
    id: ADMIN_ID,
    name: "user1",
    email: "user1@gmail.com",
    password: "user@123123",
    remember: true,
    agreeTerms: true,
    pinCode: "118791",
    role: USER_ROLE.ADMIN,
    gender: 0,
    location: "112 Nguyen Thi Dinh, Da Nang",
    phoneNumber: "(+84)763651133",
    birthday: "2024-04-02T04:38:38.648",
    language: "English",
    skills: ["0", "1"],
    avatar:
      "https://demos.creative-tim.com/nextjs-material-dashboard-pro//_next/static/media/black-chair.b2719b4f.jpeg",
    createdAt: "2023-11-22T04:01:46+00:00",
    bio: "This is description",
    facebookUrl: "facebook.com",
    instagramUrl: "instagram.com",
    twitterUrl: "twitter.com",
  },
  {
    id: 2,
    name: "user2",
    email: "user2@gmail.com",
    password: "user2",
    remember: false,
    agreeTerms: false,
    role: USER_ROLE.NORMAL,
    gender: 0,
    location: "112 Nguyen Thi Dinh, Da Nang",
    phoneNumber: "(+84)763651133",
    birthday: "2024-04-02T04:38:38.648",
    language: "English",
    skills: ["0", "1"],
    avatar:
      "https://demos.creative-tim.com/nextjs-material-dashboard-pro//_next/static/media/chair-pink.686a7ae1.jpeg",
    createdAt: "2023-08-08T23:55:34+00:00",
    bio: "",
    facebookUrl: "",
    instagramUrl: "",
    twitterUrl: "",
  },
];

export const MOCK_USER_BIRTHDAY: BirthDay = {
  date: 1,
  months: 1,
  years: 1990,
};

export const MOCK_EMAIL_OPTIONS = [
  {
    option: "tremor+100@gmail.com",
    value: "1",
  },
  {
    option: "tremor+200@gmail.com",
    value: "2",
  },
  {
    option: "tremor+300@gmail.com",
    value: "3",
  },
];
