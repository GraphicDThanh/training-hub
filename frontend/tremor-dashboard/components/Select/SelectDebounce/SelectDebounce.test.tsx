import { render } from "@testing-library/react";

//
import SelectDebounce from "./SelectDebounce";
import { MOCK_EMAIL_OPTIONS } from "@/mocks";

describe("SelectDebounce", () => {
  const renderComponent = () =>
    render(
      <SelectDebounce
        id="toUserId"
        label="User"
        options={MOCK_EMAIL_OPTIONS}
      />,
    );

  // TODO: I will update the onChange case later
  it("Match snapshot", () => {
    const { container } = renderComponent();

    expect(container).toMatchSnapshot();
  });
});
