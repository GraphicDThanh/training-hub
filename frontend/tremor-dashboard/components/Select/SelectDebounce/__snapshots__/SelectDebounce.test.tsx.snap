// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectDebounce Match snapshot 1`] = `
<div>
  <div
    class="w-full flex flex-col gap-0"
  >
    <label
      class="text-grey text-sm dark:text-lighter capitalize cursor-pointer"
      for="toUserId"
    >
      User
    </label>
    <div
      class="w-full min-w-[10rem] relative text-tremor-default"
    >
      <select
        class="h-full w-full absolute left-0 top-0 z-0 opacity-0"
        id="toUserId"
        title="search-select-hidden"
      >
        <option
          class="hidden"
          disabled=""
          hidden=""
          value=""
        >
          Select...
        </option>
        <option
          class="hidden"
          value="1"
        >
          tremor+100@gmail.com
        </option>
        <option
          class="hidden"
          value="2"
        >
          tremor+200@gmail.com
        </option>
        <option
          class="hidden"
          value="3"
        >
          tremor+300@gmail.com
        </option>
      </select>
      <div
        class="w-full min-w-[10rem] relative text-tremor-default text-grey dark:text-white dark:border-light dark:focus:border-white"
        data-headlessui-state=""
        id="toUserId"
      >
        <button
          aria-expanded="false"
          aria-haspopup="listbox"
          class="w-full"
          data-headlessui-state=""
          id="headlessui-combobox-button-:r0:"
          tabindex="-1"
          type="button"
        >
          <input
            aria-autocomplete="list"
            aria-expanded="false"
            class="w-full outline-none text-left whitespace-nowrap truncate rounded-tremor-default focus:ring-2 transition duration-100 text-tremor-default pr-14 border py-2 shadow-tremor-input focus:border-tremor-brand-subtle focus:ring-tremor-brand-muted dark:shadow-dark-tremor-input dark:focus:border-dark-tremor-brand-subtle dark:focus:ring-dark-tremor-brand-muted pl-3 placeholder:text-tremor-content dark:placeholder:text-tremor-content bg-tremor-background dark:bg-dark-tremor-background hover:bg-tremor-background-muted dark:hover:bg-dark-tremor-background-muted text-tremor-content dark:text-dark-tremor-content border-tremor-border dark:border-dark-tremor-border"
            data-headlessui-state=""
            id="headlessui-combobox-input-:r1:"
            placeholder="Select..."
            role="combobox"
            type="text"
            value=""
          />
          <div
            class="absolute inset-y-0 right-0 flex items-center pr-2.5"
          >
            <svg
              class="tremor-SearchSelect-arrowDownIcon flex-none h-5 w-5 text-tremor-content-subtle dark:text-dark-tremor-content-subtle"
              fill="currentColor"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M11.9999 13.1714L16.9497 8.22168L18.3639 9.63589L11.9999 15.9999L5.63599 9.63589L7.0502 8.22168L11.9999 13.1714Z"
              />
            </svg>
          </div>
        </button>
      </div>
    </div>
  </div>
</div>
`;
